config/opal_check_ofi.m4

```diff
@@ -81,6 +135,30 @@ AC_DEFUN([_OPAL_CHECK_OFI],[
                               [],
                               [opal_ofi_happy=no])])
 
+    dnl If we got happiness, then check the Libfabric API version.
+    AS_IF([test "$opal_ofi_happy" = "yes"], [
+        AC_PROG_CPP
+
+        AC_MSG_CHECKING([for major/minor OFI API version])
+
+        dnl The goal here is to do this test without compiling/runnning
+        dnl a program (because that is problematic in cross-compilation
+        dnl scenarios).  So extract the values we want by using the CPP.
+        dnl
+        dnl See comment above about why we call this macro twice, once
+        dnl with nothing and once with -P.
+        _OPAL_CHECK_OFI_GET_API_VERSION([])
+        AS_IF([test -z "$OFI_MAJOR_VERSION" || test -z "$OFI_MINOR_VERSION"],
+              [_OPAL_CHECK_OFI_GET_API_VERSION([-P])])
+
+        dnl Sanity check: make sure we got values
+        AS_IF([test -z "$OFI_MAJOR_VERSION" || test -z "$OFI_MINOR_VERSION"],
+              [AC_MSG_RESULT([unknown])
+               AC_MSG_WARN([unable to run the C preprocessor])
```

Comment:
This warning is misleading; we're not sure why we don't have a number, but the C preprocessor might have run fine.